src/main/java/fi/metatavu/metamind/server/keycloak/AuthenticationController.java

```diff
@@ -0,0 +1,408 @@
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.Map.Entry;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.ws.rs.InternalServerErrorException;
+import javax.ws.rs.core.Response;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.jboss.security.authorization.AuthorizationException;
+import org.keycloak.admin.client.Keycloak;
+import org.keycloak.admin.client.KeycloakBuilder;
+import org.keycloak.admin.client.resource.RealmResource;
+import org.keycloak.admin.client.resource.ResourcesResource;
+import org.keycloak.admin.client.resource.ScopePermissionsResource;
+import org.keycloak.admin.client.resource.UserPoliciesResource;
+import org.keycloak.admin.client.resource.UsersResource;
+import org.keycloak.representations.idm.ClientRepresentation;
+import org.keycloak.representations.idm.UserRepresentation;
+import org.keycloak.representations.idm.authorization.DecisionEffect;
+import org.keycloak.representations.idm.authorization.DecisionStrategy;
+import org.keycloak.representations.idm.authorization.PolicyEvaluationRequest;
+import org.keycloak.representations.idm.authorization.PolicyEvaluationResponse;
+import org.keycloak.representations.idm.authorization.ResourceOwnerRepresentation;
+import org.keycloak.representations.idm.authorization.ResourceRepresentation;
+import org.keycloak.representations.idm.authorization.ScopePermissionRepresentation;
+import org.keycloak.representations.idm.authorization.ScopeRepresentation;
+import org.keycloak.representations.idm.authorization.UserPolicyRepresentation;
+import org.slf4j.Logger;
+
+/**
+ * Class providing authentication keycloak system to evaluate permissions
+ *  
+ * @author DaniilSmirnov1
+ *
+ */
+@ApplicationScoped
+public class AuthenticationController {
+  
+  private static final String REALM = System.getProperty("keycloak-admin-realm");
+  private static final String CLIENT_ID = System.getProperty("keycloak-admin-client-id");
+  private static final String ADMIN_USER = System.getProperty("keycloak-admin-user");
+  private static final String ADMIN_PASSWORD = System.getProperty("keycloak-admin-password");
+  private static final String SERVER_URL = System.getProperty("keycloak-admin-server-url");
+  private static final String CLIENT_SECRET = System.getProperty("keycloak-admin-client-secret");
+  
+  @Inject 
+  private static Logger logger;
+  
+  /**
+   * Returns admin client builder
+   * 
+   * @return
+   */
+  private Keycloak getAdminClient() {
```

Comment:
Methods should be arranged so that public methods are first, protected next and privates last